shared/wallets/onboarding/container.tsx

```diff
@@ -41,8 +41,15 @@ const ConnectedOnboarding = Container.connect(
 // A wrapper to harmonize the type of OwnProps between the
 // routed case and <Onboarding /> case.
 type RoutedOnboardingProps = Container.RouteProps<OwnProps>
-export const RoutedOnboarding = (ownProps: RoutedOnboardingProps) => (
-  <ConnectedOnboarding nextScreen={Container.getRouteProps(ownProps, 'nextScreen', 'openWallet')} />
-)
+export const RoutedOnboarding = (ownProps: RoutedOnboardingProps) => {
+  React.useEffect(() => {
+    // @ts-ignore
+    if (!ownProps.navigation.getParam('hideHeader')) {
+      // @ts-ignore
+      ownProps.navigation.setParams({hideHeader})
```

Comment:
i don't understand what this is for. you're getting the value, then just settings it again?

Reply:
I found without the above check, setting the params seems to cause an infinite render loop – I guess setting params triggers a rerender? I was a bit surprised by this because it wasn't mentioned in the [React Navigation example](https://reactnavigation.org/docs/en/redux-integration.html#setparams-from-your-screen) for setting params from a component.